src/EditorFeatures/Test2/GoToDefinition/GoToDefinitionCommandHandlerTests.vb

```diff
@@ -88,7 +88,7 @@ class C
     </Project>
 </Workspace>
 
-            Using workspace = TestWorkspace.Create(definition, exportProvider:=GoToTestHelpers.ExportProvider)
+            Using workspace = TestWorkspace.Create(definition, exportProvider:=GoToTestHelpers.ExportProviderFactory.CreateExportProvider())
                 Dim cursorDocument = workspace.Documents.First(Function(d) d.CursorPosition.HasValue)
                 Dim cursorPosition = cursorDocument.CursorPosition.Value
 
```

Comment:
this will be the example. here it uses mock IDocumentNavigationService and IStreamFindUsagespresenter mock. 

but one of them is given as dependency and the other uses MEF. one is probably given as MEF since it is not exposed through constructor of CSharpGotoDefinitionService but want to easily inject that dependency under several layers.

and the other is happen to be exposed through CSharpGoToDefinitionService. unless I know exactly how these dependencies are used, I can't be sure whether IStreamingFindUsagesPresernter I gave in here is the one used internally.

removing all possibility to provide dependency except through MEF removes your concern that some tests uses 2 different instances of same service (one from constructor and one from MEF) and making test flaky.

especially, when one can add dependency through MEF to new service without realizing one had to change test like this as well since one just introduced possiblity of tests having 2 different instance of same service.

Reply:
by the way, I just searched Mock in roslyn solution. there are quite more mocks you used in our tests. I didn't even search for Testxxx. since I believe we used Mockxxx or Testxxx for our mock dependencies for unit tests.

Reply:
Correct, fixing the mocks is left as a follow-up item. :smile: